src/dynamic_loader.cpp

```diff
@@ -26,18 +26,18 @@
 #include <migraphx/file_buffer.hpp>
 #include <migraphx/tmp_dir.hpp>
 #include <utility>
-
 #include <dlfcn.h>
 
 namespace migraphx {
 inline namespace MIGRAPHX_INLINE_NS {
-
 struct dynamic_loader_impl
 {
     dynamic_loader_impl() = default;
+
     dynamic_loader_impl(const fs::path& p, std::shared_ptr<tmp_dir> t = nullptr)
-        : handle(dlopen(p.string().c_str(), RTLD_LAZY), &dlclose), temp(std::move(t))
+        : handle(dlopen(p.string().c_str(), RTLD_LAZY), dlclose_wrapper{}), temp(std::move(t))
     {
+        dynamic_loader::check_load_error();
```

Comment:
We should be checking the error for `dlopen` and then throw.

Reply:
When constrctor gets called, it calls into `dlopen` and then, this checks for error on `dlopen` and throws if necessary. So it is happening.